packages/generator-common/src/file-writer/create-file.ts

```diff
@@ -1,9 +1,138 @@
 
-const { writeFile } = promises;
+const { writeFile, readFile } = promises;
+const logger = createLogger('create-file');
+
+/**
+ * Options for the creation of a file.
+ * @internal
+ */
+export interface CreateFileOptions {
+  /**
+   * Flag to indicate if the file is overwritten.
+   */
+  overwrite?: boolean;
+  /**
+   * Flag to indicate if copy write header is added.
+   */
+  withCopyright?: boolean;
+  /**
+   * Path to the prettier config with respect to the process.cwd().
+   */
+  prettierConfigPath?: string;
+  /**
+   * Flag to indicate if the file is prettified before.
+   */
+  usePrettier?: boolean;
+}
+
+/**
+ * @internal
+ */
+export const defaultPrettierConfig: PrettierOptions = {
+  singleQuote: true,
+  trailingComma: 'none',
+  arrowParens: 'avoid',
+  endOfLine: 'lf'
+};
+
+let prettierConfigCache: PrettierOptions | undefined;
+
+/**
+ * @internal
+ */
+export function clearPrettierConfigCache(): void {
+  prettierConfigCache = undefined;
+}
+
+async function readPrettierConfig(
+  prettierConfigPath: string | undefined
+): Promise<PrettierOptions> {
+  if (prettierConfigCache) {
+    return prettierConfigCache;
+  }
+
+  if (prettierConfigPath) {
+    const resolved = resolve(process.cwd(), prettierConfigPath);
+    try {
+      const config = await readFile(resolved, { encoding: 'utf-8' });
+      prettierConfigCache = JSON.parse(config);
+    } catch (e) {
+      logger.warn(
+        `Prettier config file not found: ${resolved} - default is used.`
+      );
+      prettierConfigCache = defaultPrettierConfig;
+    }
+  }
+
+  return prettierConfigCache || defaultPrettierConfig;
+}
+
+const fileParserMap: Record<string, BuiltInParserName> = {
+  ts: 'typescript',
+  md: 'markdown',
+  json: 'json',
+  js: 'espree',
+  mdx: 'mdx',
+  yml: 'yaml',
+  yaml: 'yaml',
+  'd.ts': 'typescript',
+  'js.map': 'json',
+  'd.ts.map': 'json'
+};
+
+/**
+ * This method considers also double dots like `.map.js`
+ * @param fileName
+ * @returns The complete file extension containing multiple dots
+ * @internal
+ */
+export function getFileExtension(fileName: string): string {
+  const fileType = parse(fileName).base.split('.').slice(1).join('.');
+  return fileType;
+}
+
+async function formatWithPrettier(
+  fileName: string,
+  content: string,
+  prettierConfigPath: string | undefined
+) {
+  const prettierConfig = await readPrettierConfig(prettierConfigPath);
+  const fileExtension = getFileExtension(fileName);
+  const parser: BuiltInParserName | undefined = fileParserMap[fileExtension];
+
+  if (parser) {
+    return prettier.format(content, { ...prettierConfig, parser });
+  }
+  logger.info(
+    `No prettier-parser configured for file ${fileName} - skip prettier.`
+  );
+  return content;
+}
+
+function addCopyrightHeader(content: string, withCopyright: boolean): string {
+  if (!withCopyright) {
+    return content;
+  }
+  return (
+    codeBlock`
+${getCopyrightHeader()}
+${content}
+` + unixEOL
```

Comment:
[req] I would put the `unixEOL` into the codeblock.
```suggestion
${content}
${unixEOL}
`
```
or:

```suggestion
${content}\n
`
```
[pp] After using it for a while, I see that I am not a fan of the `unixEOL`.  `\n` is just shorter, does not require escaping in strings and you don't need to import it. Maybe we could reconsider this.